src/main/java/attendance/domain/CrewHistories.java

```diff
@@ -1,31 +1,95 @@
+        this.histories = new HashMap<>(histories);
     }
 
-    public CrewHistory findCrewByNickname(final String nickname) {
-        if (!crews.containsKey(nickname)) {
+    public void addHistory(final String nickname, final LocalDateTime attendanceDateTime) {
+        createIfNotExists(nickname);
+        CrewHistory crewHistory = histories.get(nickname);
+        crewHistory.add(attendanceDateTime);
+    }
+
+    public void validateHistoryNotExists(final String nickname, final LocalDate attendanceDate) {
+        validateKeyExists(nickname);
+        CrewHistory crewHistory = histories.get(nickname);
+        crewHistory.validateNotExists(attendanceDate);
+    }
+
+    public void validateHistoryExists(final String nickname, final LocalDate attendanceDate) {
+        validateKeyExists(nickname);
+        CrewHistory crewHistory = histories.get(nickname);
+        crewHistory.validateExists(attendanceDate);
+    }
+
+    public void validateKeyExists(final String nickname) {
+        if (!histories.containsKey(nickname)) {
             throw new IllegalArgumentException("[ERROR] 등록되지 않은 닉네임입니다.");
         }
-        return crews.get(nickname);
     }
 
-    public Map<String, AttendanceCounter> findDismissalCrews(final LocalDate todayDate) {
-        return crews.entrySet().stream()
-                .map(entry -> Map.entry(entry.getKey(), entry.getValue().countAttendanceStatus(todayDate)))
-                .filter(entry -> WarningLevel.from(entry.getValue().getAbsentCount(), entry.getValue().getLateCount())
-                        .isApplicable())
-                .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+    public LocalDateTime modify(final String nickname, final LocalDateTime modifyingDateTime, final LocalDate nowDate) {
+        validateKeyExists(nickname);
+        validatePreviousDate(LocalDate.from(modifyingDateTime), nowDate);
+        CrewHistory crewHistory = histories.get(nickname);
+        return crewHistory.modify(modifyingDateTime);
+    }
+
+    public Optional<LocalDateTime> findHistoryOfDate(final String nickname, final LocalDate date) {
+        CrewHistory crewHistory = histories.get(nickname);
+        return crewHistory.find(date);
+    }
+
+    public CrewHistory findHistory(final String nickname) {
+        validateKeyExists(nickname);
+        return histories.get(nickname);
+    }
+
+    public Map<String, AttendanceCounter> makeAttendanceCounterByNickname(final LocalDate nowDate,
+                                                                          final CampusScheduler campusScheduler) {
+        Map<String, AttendanceCounter> result = new HashMap<>();
+        for (Entry<String, CrewHistory> entry : histories.entrySet()) {
+            CrewHistory history = entry.getValue();
+            AttendanceCounter counter = campusScheduler.countByAttendanceState(history, nowDate);
+            result.put(entry.getKey(), counter);
+        }
+        return result;
```

Comment:
```suggestion
        return histories.entrySet().stream()
                .collect(Collectors.toMap(
                        Entry::getKey,
                        entry -> campusScheduler.countByAttendanceState(entry.getValue(), nowDate)
                ));
```

Stream을 이용하면 이렇게 표현해볼 수 있습니다 :)

Reply:
`Map`과 관련된 `stream` 메서드를 공부해보겠습니다! 감사합니다 !!